doc/modules/array_api.rst

```diff
@@ -101,6 +101,7 @@ Estimators
 - :class:`preprocessing.MaxAbsScaler`
 - :class:`preprocessing.MinMaxScaler`
 - :class:`preprocessing.Normalizer`
+- :class:`preprocessing.StandardScaler`
```

Comment:
I think we should add a special marker to estimators that have a device dependent dtype handling (and therefore numerical precision guarantees). For instance we could have a link such as:

```suggestion
- :class:`preprocessing.StandardScaler` (see :ref:`float64_support`)
```

that points to a new section with that label that explains that some namespaces / device combinations such as torch with the MPS device do not support `float64` operations while the estimators usually to upcast some operations to higher precision when fed with `float32` data. For devices where this is not possible, the computation stays in `float32` at the cost of potential numerical precision problems.

Reply:
Note that the `float64` upcast was introduced in #12338 without much discussion on the motivation and possible alternatives.

In particular, if we were to make the columns of `X` contiguous (maybe one column at a time to avoid a full input data copy), `numpy.(nan)sum` would have higher numerical precision via the use of pairwise summation internally (both for float32 and float64 data) as explained in https://github.com/numpy/numpy/issues/9393#issuecomment-313925335.

However I don't want to change the current behavior of `StandardScaler` for numpy by the introduction of Array API support, so I don't a good alternative to the degraded numerical precision of `StandardScaler` on namespace/device combos that do not support `float64`.